deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/ProtonJExceptionParser.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (c) Microsoft. All rights reserved.
+ *  Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+package com.microsoft.azure.sdk.iot.deps.transport.amqp;
+
+import lombok.Getter;
+import org.apache.qpid.proton.amqp.transport.ErrorCondition;
+import org.apache.qpid.proton.engine.Endpoint;
+import org.apache.qpid.proton.engine.Event;
+
+@Getter
+public class ProtonJExceptionParser
+{
+    private String error;
+    private String errorDescription;
+
+    private static final String DEFAULT_ERROR_DESCRIPTION = "NoErrorDescription";
+
+    public ProtonJExceptionParser(Event event)
+    {
+        getTransportExceptionFromProtonEndpoints(event.getSender(), event.getReceiver(), event.getConnection(), event.getTransport(), event.getSession(), event.getLink());
+    }
+
+    private ErrorCondition getErrorConditionFromEndpoint(Endpoint endpoint)
+    {
+        return endpoint.getCondition() != null && endpoint.getCondition().getCondition() != null ? endpoint.getCondition() : endpoint.getRemoteCondition();
+    }
+
+    private void getTransportExceptionFromProtonEndpoints(Endpoint... endpoints)
+    {
+        for (Endpoint endpoint : endpoints)
+        {
+            if (endpoint == null)
```

Comment:
Why is this check required? It would be better to check that `endpoints` is not null.

Reply:
Good point, this check isn't doing anything so I will remove it

Reply:
This check is required because that collection of endpoints does contain some null endpoints

Reply:
The collection of endpoints is never null, though

Reply:
The `Endpoint` spread is over the inputs we get from the constructor; makes sense.